crates/forge/tests/cli/revive_config.rs

```diff
@@ -0,0 +1,45 @@
+//! Contains tests for checking configuration for zksync
+
+use foundry_test_utils::util::OTHER_SOLC_VERSION;
+
+pub const OTHER_REVIVE_VERSION: &str = "revive:0.1.0-dev.12";
+
+// tests that `--use <solc>` works
+forgetest!(can_use_solc, |prj, cmd| {
+    prj.add_raw_source(
+        "Foo",
+        r"
+pragma solidity *;
+contract Foo {}
+   ",
+    )
+    .unwrap();
+
+    cmd.args(["build", "--use", OTHER_SOLC_VERSION, "--revive"]).assert_success().stdout_eq(str![
+        [r#"
+[COMPILING_FILES] with [REVIVE_VERSION]
+[REVIVE_VERSION] Finished [ELAPSED]
+Compiler run successful!
+
+"#]
+    ]);
+
+    cmd.forge_fuse().args([
+        "build",
+        "--force",
+        "--revive",
+        "--use-revive",
+        &format!("revive:{OTHER_REVIVE_VERSION}"),
+    ]);
+    cmd.assert_failure().stderr_eq(str![[r#"
+Error: `revive` selecting by versions is not supported
+
+"#]]);
+
+    // fails to use solc that does not exist
```

Comment:
solc?

Reply:
thanks, should be revive or resolc. I do not get why we need to different names for compiler and binary

Reply:
i'd leave it as it is for now and would change to use `resolc` everywhere. As per convention for compiler wrapping to use explicit naming.